packages/flutter_test/lib/src/controller.dart

```diff
@@ -271,6 +314,321 @@ class SemanticsController {
 
     return false;
   }
+
+  /// Recursive depth first traversal of the specified `node`, adding nodes
+  /// that are important for semantics to the `traversal` list.
+  void _accessibilityTraversal(SemanticsNode node, List<SemanticsNode> traversal){
+    if (_isImportantForAccessibility(node)) {
+      traversal.add(node);
+    }
+
+    final List<SemanticsNode> children = node.debugListChildrenInOrder(DebugSemanticsDumpOrder.traversalOrder);
+    for (final SemanticsNode child in children) {
+      _accessibilityTraversal(child, traversal);
+    }
+  }
+
+  /// Performs the given [SemanticsAction] on the [SemanticsNode] found by `finder`.
+  ///
+  /// If `args` are provided, they will be passed unmodified with the `action`.
+  /// The `checkForAction` argument allows for attempting to perform `action` on
+  /// `node` even if it doesn't report supporting that action. This is useful
+  /// for implicitly supported actions such as [SemanticsAction.showOnScreen].
+  void performAction(FinderBase<SemanticsNode> finder, SemanticsAction action, {Object? args, bool checkForAction = true}) {
+    final SemanticsNode node = finder.evaluate().single;
+    if (checkForAction && !node.getSemanticsData().hasAction(action)){
+      throw StateError(
+        'The given node does not support $action.\n'
```

Comment:
Maybe mention the `checkForAction` flag in the error?

Reply:
Added.